torch/_dynamo/utils.py

```diff
@@ -299,6 +299,159 @@ def get_runtime_metrics_context() -> RuntimeMetricsContext:
     return _RUNTIME_METRICS_CONTEXT
 
 
+class CompileEventLogLevel(enum.Enum):
+    """
+    Enum that loosely corresponds with a "log level" of a given event.
+
+    CHROMIUM_EVENT: Logs only to tlparse.
+    COMPILE_EVENT: Logs to tlparse + PT2 Compile Events
+    COMPILATION_METRIC: Logs to tlparse, PT2 Compile Events, and dynamo_compile
+    """
+
+    CHROMIUM = 1
+    PT2_COMPILE = 2
+    COMPILATION_METRIC = 3
+
+
+class CompileEventLogger:
+    """
+    Helper class for representing adding metadata(i.e. columns) to various compile events.
+    Use CompileEventLogger.add_data() to add metadata to:
+    - Chromium events
+    - PT2 Compile Events
+    - CompilationMetrics
+
+    This should be used in conjunction with dynamo_timed(), which creates the various
+    events and contexts to add data to.
+    """
+
+    @staticmethod
+    def log_instant_event(
+        event_name: str,
+        log_level: CompileEventLogLevel,
+        metadata: Dict[str, object],
+        time_ns: Optional[int] = None,
+    ):
+        if time_ns is None:
+            time_ns = time.time_ns()
+        chromium_log = get_chromium_event_logger()
+        if log_level == CompileEventLogLevel.CHROMIUM:
+            log_pt2_compile_event = False
+        elif log_level == CompileEventLogLevel.PT2_COMPILE:
+            log_pt2_compile_event = True
+        else:
+            raise RuntimeError(
+                "Cannot log instant event at COMPILATION_METRIC level. Please choose one of CHROMIUM_EVENT or COMPILE_EVENT"
+            )
+        chromium_log.log_instant_event(
+            event_name, time_ns, metadata, log_pt2_compile_event
+        )
+
+    @staticmethod
+    def add_data_(event_name: str, log_level: CompileEventLogLevel, **metadata: object):
+        """
+        Centralized API for adding data to various events
+        Log an event to a toplevel "dynamo" event or metrics context
+        depending on log level.
+        """
+        chromium_log = get_chromium_event_logger()
+        pt2_compile_substack = chromium_log.get_pt2_compile_substack()
+
+        if log_level == CompileEventLogLevel.CHROMIUM:
+            chromium_log.add_event_data(event_name, **metadata)
+        elif log_level == CompileEventLogLevel.PT2_COMPILE:
+            pt2_compile_substack = chromium_log.get_pt2_compile_substack()
```

Comment:
I don't know what's going on in this branch. It looks like you already did `pt2_compile_substack = chromium_log.get_pt2_compile_substack()` above. Also, since this is a different log_level so to speak, I was expecting to see some additional logging. But it doesn't look like you're logging anything different in this branch -- just checking event name is in `pt2_compile_substack`. Does the additional logging to pt2_compile happen in `chromium_log.add_event_data()`? If so, I'd expect this check to happen there.

Reply:
Basically, if you pass in PT2_COMPILE as the log level, you are asserting that this event needs to be logged to Scuba, i.e. the dynamo_timed() you're calling has `log_pt2_compile_event=True`.  This substack check is just the easiest way of checking that. 

Therefore, the event that it's adding to should be part of the pt2_compile stack, or else something has gone wrong. Technically, you could just not check this assertion, but then we will silently not log anything to Scuba, which seems worse.